memcache.go

```diff
@@ -192,17 +184,22 @@ func UnicodeItem(k, s string) *memcache.Item {
 	}
 }
 
-// BoolItem returns a memcache.Item sutable for storing a boolean
+// BoolItem returns a memcache.Item suitable for storing a boolean
 // this provides compatability with pylibmc
+// to maintain compatibility between python2 and python3,
+// the values are pickled as True or False, rather than 1 or 0
+// In turn, go will unpickle this value whenever it is set.
 func BoolItem(k string, v bool) *memcache.Item {
-	value := "0"
+	// pickled boolean of False
+	value := []byte("\x80\x02\x89.")
 	if v {
-		value = "1"
+		// pickled boolean of True
+		value = []byte("\x80\x02\x88.")
 	}
 	return &memcache.Item{
 		Key:   k,
-		Value: []byte(value),
-		Flags: FLAG_BOOL,
+		Value: value,
+		Flags: FLAG_NONE,
 	}
 }
```

Comment:
I don't think this is backwards compatible with Go code running the existing code because it only handles pickled unicode string and would not be able to parse values set by this new code.